nixos/modules/services/backup/borgbackup.nix

```diff
@@ -78,8 +76,13 @@ let
       ];
       script = mkBackupScript cfg;
       serviceConfig = {
-        User = cfg.user;
-        Group = cfg.group;
+        ExecStartPre = "+${pkgs.writeShellScript "borgbackup-pre-${name}" cfg.preHook}";
+        ExecStopPost = "+${pkgs.writeShellScript "borgbackup-post-${name}" cfg.postHook}";
```

Comment:
post hooks are also used for monitoring. Is there a way to still get `exitStatus` in this case? I currently do this: https://github.com/Mic92/dotfiles/blob/e22911ae4d929bc91316a2c4a4112ad175b5bc77/nixos/matchbox/modules/borgbackup.nix#L34

Reply:
>  Is ExecStopPost executed in both the error and success case?

Yes, it is a cleanup job basically.

> post hooks are also used for monitoring. Is there a way to still get exitStatus in this case? 

I suspect we could move that use case into `ExecStartPost`. This would obviously entail that we change the nomenclature on the hooks a bit.

Reply:
Looks like `EXIT_STATUS` is what you are looking for.

```
client # [   36.370212] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: extraArgs=
client # [   36.373147] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: LANG=en_US.UTF-8
client # [   36.375511] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: SERVICE_RESULT=success
client # [   36.378141] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: extraPruneArgs=
client # [   36.380297] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: INVOCATION_ID=2cc515adada14ffc99b45965e1c43319
client # [   36.381879] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: EXIT_STATUS=0
client # [   36.383178] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: BORG_REPO=borg@server:.
client # [   36.384531] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: USER=borgbackup-job-remote
client # [   36.386276] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: TZDIR=/nix/store/yijklpbih24arzw7jcc3ykwbndd8gbgf-tzdata-2020f/share/zoneinfo
client # [   36.388193] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: PWD=/
client # [   36.389428] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: JOURNAL_STREAM=8:13995
client # [   36.390853] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: BORG_BASE_DIR=/var/lib/borgbackup
client # [   36.392377] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: extraInitArgs=
client # [   36.399521] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: extraCreateArgs=
client # [   36.401899] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: STATE_DIRECTORY=/var/lib/borgbackup
client # [   36.404020] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: SHLVL=1
client # [   36.405827] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: EXIT_CODE=exited
client # [   36.407573] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: LOCALE_ARCHIVE=/nix/store/hvjscb69218cw0r60pg7szxv8d5dk6lz-glibc-locales-2.32-40/lib/locale/locale-archive
client # [   36.410466] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: LOGNAME=borgbackup-job-remote
client # [   36.413138] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: BORG_RSH=ssh -oStrictHostKeyChecking=no -i /root/id_ed25519
client # [   36.415573] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: PATH=/nix/store/01xwxvlyavk7n8gpmmjbryrjflgs4cl9-borgbackup-1.1.16/bin:/nix/store/6izw7xiy7f7lib7dsqn2ip476kmpkzm5-openssh-8.5p1/bin:/nix/store/lr96h3dlny8aiba9p3rmxcxfda0ijj08-coreutils-8.32/bin:/nix/store/6kjrp5p21icxr38zws6yzbdsbf8z66b6-findutils-4.7.0/bin:/nix/store/3v5i98i92j0f3lbb7d58kvf8nxnhw7s7-gnugrep-3.6/bin:/nix/store/4nf4ih03fcq7gk08spjzxvwph1vyx1kr-gnused-4.8/bin:/nix/store/ixpwj1cxl4rp2qbyn0s2h4k5zj731q7c-systemd-247.6/bin:/nix/store/01xwxvlyavk7n8gpmmjbryrjflgs4cl9-borgbackup-1.1.16/sbin:/nix/store/6izw7xiy7f7lib7dsqn2ip476kmpkzm5-openssh-8.5p1/sbin:/nix/store/lr96h3dlny8aiba9p3rmxcxfda0ijj08-coreutils-8.32/sbin:/nix/store/6kjrp5p21icxr38zws6yzbdsbf8z66b6-findutils-4.7.0/sbin:/nix/store/3v5i98i92j0f3lbb7d58kvf8nxnhw7s7-gnugrep-3.6/sbin:/nix/store/4nf4ih03fcq7gk08spjzxvwph1vyx1kr-gnused-4.8/sbin:/nix/store/ixpwj1cxl4rp2qbyn0s2h4k5zj731q7c-systemd-247.6/sbin
client # [   36.428538] 9ld91vi1kc745zqc0g3fndc1a9x7db2z-borgbackup-post-remote[903]: _=/nix/store/lr96h3dlny8aiba9p3rmxcxfda0ijj08-coreutils-8.32/bin/env
```

Reply:
Documented here: https://www.freedesktop.org/software/systemd/man/systemd.exec.html#%24EXIT_CODE